interface/admin.py

```diff
@@ -1,6 +1,46 @@
-from django.contrib import admin
+from zipfile import ZipFile
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
+from django.contrib import admin, messages
+from django.http import FileResponse
+
 from interface.models import Course, Assignment, Submission
 
 admin.site.register(Course)
 admin.site.register(Submission)
-admin.site.register(Assignment)
+
+
+@admin.register(Assignment)
+class AssignmentAdmin(admin.ModelAdmin):
+    actions = ['download_submissions']
+
+    def download_submissions(self, request, queryset):
+        if queryset.count() != 1:
+            messages.warning(request, 'Only one assignment can be selected')
+            return request
+
+        assignment = queryset[0]
+        submission_set = assignment.submission_set.order_by('timestamp')
+
+        submissions = {}
+
+        # we only want the last submission of every user
+        for submission in submission_set:
+            submissions[submission.user.username] = submission
+
+        with TemporaryDirectory() as _tmp:
+            tmp = Path(_tmp)
+
+            with ZipFile(tmp / 'review.zip', 'x') as zipfile:
+                for submission in submissions.values():
+                    submission.download(tmp / f'{submission.id}.zip')
+                    zipfile.write(
+                        tmp / f'{submission.id}.zip',
+                        f'{submission.user.username}-{submission.id}.zip'
+                        )
```

Comment:
Please unindent the closing bracket. Flake8 says `41 col 25| E123 closing bracket does not match indentation of opening bracket's line`. Also, add a final comma to the last argument of the function.

Reply:
It's strange because the CI didn't cry about it.